### ISSUE TITLE
flavor_text_entries contain newline and other characters within the string,

### ISSUE THREAD

**Issue by tanxh33** [PokeAPI/pokeapi#719]
### Summary
- The [source data](https://github.com/veekun/pokedex/blob/master/pokedex/data/csv/) for `flavor_text` uses characters like `\n` and `\u000c` which make it difficult to read normally
- PokéAPI contains the same characters which might not be desirable
- Suggest to parse the raw text into a readable format for the benefit of most common use-cases
- If not, consider adding this info into the docs so users might have a chance to know about it and handle the data appropriately

---

### Problem

Some `flavor_text_entries` have issues with additional unexpected characters.

Example from [Wobbuffet's entry](https://pokeapi.co/api/v2/pokemon-species/wobbuffet):

```json
"flavor_text_entries" : [
{ "flavor_text" : "It hates light and\nshock. If attack­\ned, it inflates\u000cits body to pump\nup its counter­\nstrike." }
...
{ "flavor_text" : "To keep its pitch-\nblack tail hidden,\nit lives quietly\u000cin the darkness.\nIt is never first\nto attack." }
...
{ "flavor_text" : "In order to con­\nceal its black\ntail, it lives in\u000ca dark cave and\nonly moves about\nat night." }
...
]
```

Seems like it is also not limited to `PokemonSpecies`, nor to language. See the move [Seismic Toss](https://pokeapi.co/api/v2/move/69):

```json
"Inflicts damage identical\nto the user’s level."
"L’ennemi est projeté grâce au pouvoir de\nla gravité. Inflige des dégâts équivalents\nau niveau du lanceur."
"いんりょくを　つかい　なげとばす。\nじぶんの　レベルと　おなじ　ダメージを\nあいてに　あたえる。"
```

### Details of the problem

These characters seem to originate from the [CSV file data from veekun/pokedex](https://github.com/veekun/pokedex/blob/master/pokedex/data/csv/). A comment on this closed issue about "[parsing pokemon_species_flavor_text.csv](https://github.com/veekun/pokedex/issues/218#issuecomment-339841781)" confirms that the presence of these characters within the files **is the intended behaviour**, to reflect the original format of the text within the game files, which contain "explicit line breaks, hyphenation, and page breaks".

There is also a suggested method to replace these characters to make it more readable, although it is a *little* more complicated than simply "replace `\n` with whitespace/empty string".

### Impact and Suggestions

<img src="https://user-images.githubusercontent.com/54199648/169378325-b693a4c3-441f-41ec-ba8a-b242bb9d9cf6.png" height="200">

Above is an example of someone's application using the unprocessed `flavor_text` directly. which has an unfortunate result of making it look somewhat unrefined. It's an avoidable outcome which the PokéAPI team can perhaps help mitigate.

The team may want to implement parsing of the `flavor_text` for the API, since I would expect that most users of the API would simply take the `flavor_text` as it is, assuming that it is formatted correctly.

Alternatively, if it is neither feasible nor desirable to do this parsing on the API side, then perhaps it should be stated within the documentation that the `flavor_text` are in an unprocessed form which will require parsing on the user's side. As far as I can tell, this information is currently not being reflected in the docs. If it was, then at least some users would take note and be able to process the data on their end.

**Comment by DDriggs00** [comment]
I agree with this change. Formatting should be done by the client, not hardcoded into the data.

**Comment by Selim042** [comment]
Anyone who wants it authentically spaced would want it as is, otherwise it is relatively simple to strip all newline characters from the string.  It is minimal processing before presenting it to the user.

**Comment by merfed** [comment]
It's not authentically spaced. The flavor text is designed to function as fake paragraphs. They shouldn't be using unicode space characters in the place of actual spaces. There are improvements that can be made here.
